utils/retrieve-state/src/lib.rs

```diff
@@ -463,10 +463,7 @@ pub async fn download_trie_channels(
                 }) => {
                     let mut missing_trie_descendants = tokio::task::block_in_place(|| {
```

Comment:
This tool now needs to be rewritten to store the trie in the right order, too.
Actually, I don't understand how this still compiles: We're calling `len()` on this below.

Reply:
I've left that tool out for now, as this is still a draft (it was intended as a quick PoC for potential 1.5.0 inclusion).